lib/CAPI/Dialect/OM.cpp

```diff
@@ -171,15 +171,23 @@ MlirAttribute omEvaluatorValueGetPrimitive(OMEvaluatorValue evaluatorValue) {
 
 /// Get the Primitive from an EvaluatorValue, which must contain a Primitive.
 OMEvaluatorValue omEvaluatorValueFromPrimitive(MlirAttribute primitive) {
+  auto attr = unwrap(primitive);
+  // Create om::IntegerAttr for any mlir::IntegerAttr.
```

Comment:
I don't think we need to do this conversion. It seems like this sort of logic shouldn't belong in the C API. What was the motivation for this part? We have conversion passes to make sure we're using the right types of data by the time we get to the evaluator.

Reply:
I think this conversion is required if we want to send integer constants from evaluator. Otherwise there will be type mismatch. The other alternative would be to create `om.integer` constants in the evaluator script, instead of 
`obj = evaluator.instantiate("Test", 42) `
For example,
```
 Evaluator (INFO) actual parameter: 42 : i64
 Evaluator (INFO) format parameter type: '!om.integer'
```

Reply:
I see, thanks for clarifying. But, I still think the C API is not the right place for the conversion. I think we should generally avoid any logic in the C API layer besides converting to and from C++ objects. Could we do this in the Python code behind the `instantiate` API that wraps integers to always wrap them as om.integer attributes?

Reply:
Thanks Mike, that makes sense, I will try that.

Reply:
Yes, I think the right place would be to add the similar logic to dialects/om.py:unwrap_python_object.